domehunter/dome_control.py

```diff
@@ -0,0 +1,674 @@
+        and increment or decrement the encoder_count instance variable,
+        depending on the current rotation direction of the dome.
+
+        If the current dome direction cannot be determined, the last recorded
+        direction is adopted.
+        """
+        print(f"Encoder activated _increment_count")
+        self._turn_led_on(leds=['input_1'])
+        time.sleep(0.01)
+        self._turn_led_off(leds=['input_1'])
+        if self.current_direction == "CW":
+            self.encoder_count += 1
+        elif self.current_direction == "CCW":
+            self.encoder_count -= 1
+        # I'm unsure if this is the best way to handle a situation like this
+        elif self.current_direction is None:
+            if self.last_direction == "CW":
+                self.encoder_count += 1
+            elif self.last_direction == "CCW":
+                self.encoder_count -= 1
+
+    def _az_to_ticks(self, az):
+        """
+        Convert degrees (azimuth) to equivalent in encoder tick count.
+
+        Parameters
+        ----------
+        az : float
+            Dome azimuth position in degrees.
+
+        Returns
+        -------
+        float
+            Returns azimuth position to corresponding encoder tick count.
+
+        """
+        return az / self.az_per_tick
+
+    def _ticks_to_az(self, ticks):
+        """
+        Convert encoder tick count to equivalent in degrees (azimuth).
+
+        Parameters
+        ----------
+        ticks : integer
+            The number of encoder ticks recorded.
+
+        Returns
+        -------
+        float
+            The corresponding dome azimuth position in degrees.
+
+        """
+        return ticks * self.az_per_tick
+
+    def _move_cw(self):
+        """
+        Set dome to move clockwise.
+
+        Returns
+        -------
+        integer
+            Command status return code (tbd).
+
+        """
+        # if testing, deactivate the home_sernsor_pin to simulate leaving home
+        if self.testing and self._at_home:
+            self.home_sensor_pin.drive_low()
+        # update the last_direction instance variable
+        self.last_direction = self.current_direction
+        # now update the current_direction variable to CW
+        self.current_direction = "CW"
+        # set the direction relay switch to CW position
+        self.direction_CW_relay.on()
+        # update the debug LEDs
+        if self.last_direction == "CCW":
+            self._turn_led_off(leds=['relay_2_normally_closed'])
+        self._turn_led_on(leds=['relay_2_normally_open'])
+        # turn on rotation
+        self.rotation_relay.on()
+        self._dome_moving = True
+        # update the rotation relay debug LEDs
+        self._turn_led_on(leds=['relay_1_normally_open'])
+        self._turn_led_off(leds=['relay_1_normally_closed'])
+        cmd_status = True
+        return cmd_status
+
+    def _move_ccw(self):
```

Comment:
Lots of repetition between the two movement methods that could probably be cleaned up in some clever manner. Could be left to a future Issue.